raiden/storage/serialization/serializer.py

```diff
@@ -44,16 +48,31 @@ def serialize(obj):
         # Default, in case this is not a dataclass
         data = obj
         if is_dataclass(obj):
-            schema = SchemaCache.get_or_create_schema(obj.__class__)
-            data = schema.dump(obj)
+            try:
+                schema = SchemaCache.get_or_create_schema(obj.__class__)
+                data = schema.dump(obj)
+            except (TypeError, ValidationError) as ex:
+                raise SerializationError(f"Can't serialize: {data}") from ex
+        elif not isinstance(obj, Mapping):
+            raise SerializationError(f"Can only serialize dataclasses or dict-like objects: {obj}")
         return data
 
     @staticmethod
     def deserialize(data):
+        """ Deserialize a dict-like object.
+
+        If the key ``_type`` is present, import the target and deserialize via Marshmallow.
+        Raises ``SerializationError`` for invalid inputs.
+        """
+        if not isinstance(data, Mapping):
+            raise SerializationError(f"Can't deserialize non dict-like objects: {data}")
         if "_type" in data:
-            klass = _import_type(data["_type"])
-            schema = SchemaCache.get_or_create_schema(klass)
-            return schema.load(deepcopy(data))
+            try:
+                klass = _import_type(data["_type"])
+                schema = SchemaCache.get_or_create_schema(klass)
+                return schema.load(deepcopy(data))
+            except (ValueError, TypeError, ValidationError) as ex:
```

Comment:
It looks like the `TypeError` can only be raised by `_import_type`, could you please narrow the `try` block?